File: lab/javascript/senha_e_criptografia/verificacao_simples.js
const senha_armazenada = "1234";
const prompt = require('prompt-sync')();


function verificarSenha(senha) {
    return senha === senha_armazenada;
}

function main(){
    const MAX_TENTATIVAS = 3;
    let tentativas = 0;

    while (tentativas < MAX_TENTATIVAS) {
        const senha_usuario = prompt("Digite a senha: ");
        if (verificarSenha(senha_usuario)) {
            console.log("Acesso concedido.");
            break;
        }
        console.log("Senha incorreta. Tente novamente.");
        tentativas++;
    }
    if (tentativas === MAX_TENTATIVAS) {
        console.log("Número máximo de tentativas atingido. Acesso bloqueado.");
    }
}

main();

File: lab/c/senhas_e_criptografia/verificacao_simples.c
#include <stdio.h>
#include <string.h>

#define MAX 3
#define SENHA "1234"

int veririca_senha(char *senha) {
    if (strcmp(senha, SENHA) == 0) {
        return 1; // Senha correta
    }
    return 0; // Senha incorreta
}

int main() {
    char senha[150];
    int tentativas = 0;

    while (tentativas < MAX) {
        printf("Digite a senha: ");
        scanf("%19s", senha); // Limita a entrada para evitar overflow

        if (veririca_senha(senha)) {
            printf("Acesso permitido!\n");
            return 0; // Sai do programa com sucesso
        } else {
            printf("Senha incorreta. Tente novamente.\n");
            tentativas++;
        }
    }

    printf("Numero maximo de tentativas atingido. Acesso bloqueado.\n");
    return 1; // Sai do programa com erro
}
